src/main/java/com/minecolonies/api/tileentities/storageblocks/IStorageBlockInterface.java

```diff
@@ -0,0 +1,117 @@
+package com.minecolonies.api.tileentities.storageblocks;
+
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.crafting.ItemStorage;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+public interface IStorageBlockInterface
+{
+    /**
+     * Whether the storageblock should be included in building containers automatically.
+     *
+     * @param building The building that is being constructed.
+     * @return Whether the storageblock should be included in building containers automatically.
+     */
+    boolean shouldAutomaticallyAdd(final IBuilding building);
+
+    /**
+     * Sets whether the block is part of a warehouse.
+     *
+     * @param inWarehouse Whether it's in a warehouse
+     */
+    void setInWarehouse(final boolean inWarehouse);
```

Comment:
I feel like there should be a better way to do this, rather than leaking unnecessary methods for blocks stored in the warehouse.
There's a very specific implementation behind these methods that all storage blocks would have to follow, which makes me think this is an ill-suited method to exist here.

Reply:
As far as I can tell. This only flips a boolean `inWarehouse` which is never used anywhere. Am I missing something?

Reply:
Yeah. I think I'll actually move all of the logic to update the warehouse out of the racks. It seems like we should instead have an onChange event that the warehouse subscribes to for all of it's containers. And then the owners of the storage blocks just need to worry about firing the onChange event when appropriate. WDYT?

Reply:
Not quite, the implementation that handles this is in AbstractTileEntityRack, and it basically causes an update to every deliverable item to update their respective requests.

In another comment I suggested writing this to an abstract class, and I'd put it in there with a final method that can't be overridden.

So you have a method `setStackInSlot` which is implemented to first update the data in the rack, return a boolean if successfull.
If successfull, trigger an update to the request system with the same input stack.